.github/workflows/scheduled_code_freeze.yml

```
@@ -0,0 +1,105 @@
+name: Scheduled Code Freeze
+
+on:
+  schedule:
+    - cron: '0 21 * * FRI'  # Every Friday at 9 PM UTC
```

nit: seeing as we don't care about the _exact_ time, we can offset the minutes, to reduce the chance of github throttling it (not that we care that much, as long as it's not skipped)

```suggestion
    - cron: '3 21 * * FRI'  # Every Friday at 9:03 PM UTC
```